src/runtime/pkg/katautils/config.go

```diff
@@ -77,87 +78,88 @@ type factory struct {
+	EnableAnnotations              []string    `toml:"enable_annotations"`
+	RxRateLimiterMaxRate           uint64      `toml:"rx_rate_limiter_max_rate"`
+	TxRateLimiterMaxRate           uint64      `toml:"tx_rate_limiter_max_rate"`
+	MemOffset                      uint64      `toml:"memory_offset"`
+	DefaultMaxMemorySize           uint64      `toml:"default_maxmemory"`
+	DiskRateLimiterBwMaxRate       int64       `toml:"disk_rate_limiter_bw_max_rate"`
+	DiskRateLimiterBwOneTimeBurst  int64       `toml:"disk_rate_limiter_bw_one_time_burst"`
+	DiskRateLimiterOpsMaxRate      int64       `toml:"disk_rate_limiter_ops_max_rate"`
+	DiskRateLimiterOpsOneTimeBurst int64       `toml:"disk_rate_limiter_ops_one_time_burst"`
+	NetRateLimiterBwMaxRate        int64       `toml:"net_rate_limiter_bw_max_rate"`
+	NetRateLimiterBwOneTimeBurst   int64       `toml:"net_rate_limiter_bw_one_time_burst"`
+	NetRateLimiterOpsMaxRate       int64       `toml:"net_rate_limiter_ops_max_rate"`
+	NetRateLimiterOpsOneTimeBurst  int64       `toml:"net_rate_limiter_ops_one_time_burst"`
+	VirtioFSCacheSize              uint32      `toml:"virtio_fs_cache_size"`
+	VirtioFSQueueSize              uint32      `toml:"virtio_fs_queue_size"`
+	DefaultMaxVCPUs                uint32      `toml:"default_maxvcpus"`
+	MemorySize                     uint32      `toml:"default_memory"`
+	MemSlots                       uint32      `toml:"memory_slots"`
+	DefaultBridges                 uint32      `toml:"default_bridges"`
+	Msize9p                        uint32      `toml:"msize_9p"`
+	PCIeRootPort                   uint32      `toml:"pcie_root_port"`
+	NumVCPUs                       int32       `toml:"default_vcpus"`
+	BlockDeviceCacheSet            bool        `toml:"block_device_cache_set"`
+	BlockDeviceCacheDirect         bool        `toml:"block_device_cache_direct"`
+	BlockDeviceCacheNoflush        bool        `toml:"block_device_cache_noflush"`
+	EnableVhostUserStore           bool        `toml:"enable_vhost_user_store"`
+	VhostUserDeviceReconnect       uint32      `toml:"vhost_user_reconnect_timeout_sec"`
+	DisableBlockDeviceUse          bool        `toml:"disable_block_device_use"`
+	MemPrealloc                    bool        `toml:"enable_mem_prealloc"`
+	HugePages                      bool        `toml:"enable_hugepages"`
+	VirtioMem                      bool        `toml:"enable_virtio_mem"`
+	IOMMU                          bool        `toml:"enable_iommu"`
+	IOMMUPlatform                  bool        `toml:"enable_iommu_platform"`
+	Debug                          bool        `toml:"enable_debug"`
+	DisableNestingChecks           bool        `toml:"disable_nesting_checks"`
+	EnableIOThreads                bool        `toml:"enable_iothreads"`
+	DisableImageNvdimm             bool        `toml:"disable_image_nvdimm"`
+	HotplugVFIOOnRootBus           bool        `toml:"hotplug_vfio_on_root_bus"`
+	ColdPlugVFIO                   hv.PCIePort `toml:"cold_plug_vfio"`
```

Comment:
since `cold_plug_vfio` is a string, we should really validate what the user is passing.

Reply:
If it is not a string but some other value, toml.Decode will fail. Only on the specific key-words we will take action.

Reply:
If a user writes "rootport" and this silently does nothing, that's no good ux. I think this needs input validation.